### ISSUE TITLE
Accessibility] [Tab]  The selected tab content is not read by screen reader

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#2094]
[A11Y-623](https://jira.cengage.com/browse/A11Y-623)
[A11Y-624](https://jira.cengage.com/browse/A11Y-624)
[A11Y-625](https://jira.cengage.com/browse/A11Y-625)

**Issue 1:**
**Accessibility] [Tab] Border position > The selected tab content is not read by screen reader**

Environment:
Windows 11/Chrome 137/JAWS 2025
Windows 11/Edge 136/NVDA 2025

Steps to Reproduce:

1. Open the URL – https://react-magma.cengage.com/version/4.9.1/api/tabs/
2. Turn on the screen reader (JAWS/NVDA).
3. Navigate to the 'Tab' component page from the left navigation.
4. Navigate to "Border position" element and select different tab.
5. Verify that the selected tab content is not read.

Actual Result:

The selected tab content is not read by screen reader

Refer to screencast:
https://app.screencast.com/WVHYunyx3Q8xg

Instance:
The same issue is present for tab grouping present on the right side.
Screencast: https://app.screencast.com/wMwtsRDNDBnp0

Expected Result:

As the tab grouping is at the bottom and the content is above the tab grouping the changed content should be read by screen reader.

User Impact:
Without audible feedback, screen reader users may not realize that the tab content is displayed.

WCAG Guideline: 4.1.3 Status Messages (AA)

Bug Fix Recommendation:

a) Use an ARIA live region (e.g., aria-live="polite") to announce a confirmation message such as tab content or tab content heading
b) Make sure the live region is properly updated in the DOM for the screen reader to detect the change."

**Issue 2:**
**[Accessibility] [Tab] Border position > After selecting next and previous buttons there is no update read by screen reader.**

Environment:
Windows 11/Chrome 137/JAWS 2025
Windows 11/Edge 136/NVDA 2025

Steps to Reproduce:

1. Open the URL – https://react-magma.cengage.com/version/4.9.1/api/tabs/
2. Turn on the screen reader (JAWS/NVDA).
3. Navigate to the 'Tab' component page from the left navigation.
4. Navigate to "Scrollable " element and click forward button
5. Verify that there is no update provided to screen reader users

Actual Result:

There is no info or update message read by screen reader after user clicks forward button.

Refer to screencast:
Scrollable

Note:
This issue is also present when user clicks previous button as well.

Expected Result:

As new tabs are displayed after pressing forward button there should be update or info given to screen reader users.

User Impact:
Without audible feedback, screen reader users may not realize that there has been a change on the page.

WCAG Guideline: 4.1.3 Status Messages (AA)

Bug Fix Recommendation:

a) Use an ARIA live region (e.g., aria-live="polite") to announce a confirmation message such as "moved forward" when the reset action is performed.
b) Make sure the live region is properly updated in the DOM for the screen reader to detect the change."

**Issue 2:**
**[Accessibility] [Tab] Scrollable > After selecting next and previous buttons the focus is moving out of tab grouping**

Environment:
Windows 11/Chrome 137/JAWS 2025
Windows 11/Edge 136/NVDA 2025

Steps to Reproduce:

1. Open the URL – https://react-magma.cengage.com/version/4.9.1/api/tabs/
2. Turn on the screen reader (JAWS/NVDA).
3. Navigate to the 'Tab' component page from the left navigation.
4. Navigate to "Scrollable" element and click forward button
5. press TAB key and verify that the focus is moving out of tab grouping

Actual Result:

The TAB focus is moving out of tab grouping after pressing forward button

Refer to screencast:
https://app.screencast.com/qQBo7AxoBzKPm

Note: This issue is present for previous button as well

Expected Result:

User after clicking forward button the first tab of newly displayed tab should be focused

User Impact:
As the focus is moving out of the tab grouping after pressing forward button screen reader users will be disoriented

WCAG Guideline: 2.4.3 Focus Order (A)

Bug Fix recommendation:
Use JS function onFocus() to explicitly put focus on the first tab of newly displayed tabs.

**Extra task:**
Hey team. I'm experiencing a kind of weird bug on the [Tabs page](https://react-magma.cengage.com/version/4.12.1/api/tabs/). When it loads, it takes me halfway down the page to "Scrollable Vertical Tabs" section. Anyone else getting this same behavior?

**Comment by nikitaorliak-cengage** [comment]
**Tested on:**
- V5: https://rc--upbeat-sinoussi-f675aa.netlify.app/api/tabs/
- V4: https://next--upbeat-sinoussi-f675aa.netlify.app/api/tabs/

On V4 it’s impossible to test, `Tabs` page is broken.
On V5:
- MacOS:
Voiceover doesn’t announce content on a tab.
<img width="602" height="241" alt="Image" src="https://github.com/user-attachments/assets/bcc73893-4de8-4ab0-9d7c-ce6db6b76edd" />

- Windows:
When you move focus to the `MainPage` tab, NVDA announced the whole page.
<img width="767" height="441" alt="Image" src="https://github.com/user-attachments/assets/84d8693e-92c1-4be3-b2d6-a3728b02a51c" />

**Comment by olehnoskov-cengage** [comment]
@nikitaorliak-cengage 
Could you retest it in the scope of this [THIS](https://github.com/cengage/react-magma/pull/2381) PR?

It works well for me.
**Windows + Chrome**
https://github.com/user-attachments/assets/7a9e2015-2526-4339-bb2d-af29cd05164b

**Windows + Firefox**
https://github.com/user-attachments/assets/69b45526-bfae-4c10-9a9f-016b1b5e22c5


**MacOS + Chrome**
https://github.com/user-attachments/assets/8476efa7-8529-480b-9a9e-a76a6f1967c9

**MacOS + Safari**
https://github.com/user-attachments/assets/7355dab8-6759-4db2-b315-1bb04f85b90d

**Comment by nikitaorliak-cengage** [comment]
> [@nikitaorliak-cengage](https://github.com/nikitaorliak-cengage) Could you retest it in the scope of this [THIS](https://github.com/cengage/react-magma/pull/2381) PR?
> 
> It works well for me. **Windows + Chrome** https://github.com/user-attachments/assets/7a9e2015-2526-4339-bb2d-af29cd05164b
> 
> **Windows + Firefox** https://github.com/user-attachments/assets/69b45526-bfae-4c10-9a9f-016b1b5e22c5
> 
> **MacOS + Chrome** https://github.com/user-attachments/assets/8476efa7-8529-480b-9a9e-a76a6f1967c9
> 
> **MacOS + Safari** https://github.com/user-attachments/assets/7355dab8-6759-4db2-b315-1bb04f85b90d

Works as expected in the #2381 PR. Closing the ticket.

**Comment by olehnoskov-cengage** [comment]
> We have reverified this issue on URL: [React Magma](https://next--upbeat-sinoussi-f675aa.netlify.app/) and observed that the focus obscured behavior has been resolved. However, a new issue has been identified. Users are unable to reach the last three tabs in the group due to a focus trap between the Next and Previous buttons.

When focus reaches the Next button, pressing Tab moves focus out of the tablist. Using Shift+Tab moves focus to the Previous button, and another Shift+Tab again moves focus out of the tablist. This creates a loop between the navigation buttons, preventing access to the remaining tabs.

Please refer to the attached screen recording for reference.

Screen-recording: https://app.screencast.com/FvsmJe1BKb4NC

**Comment by olehnoskov-cengage** [comment]
After testing this ticket, please ask the audit team to retest it.
https://jira.cengage.com/browse/A11Y-599

**Comment by nikitaorliak-cengage** [comment]
Tested on:

- V5: https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/navtabs--scrolling
- V4: https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/navtabs--scrolling

Everything works as expected. No bugs found.
